test/framework/clusterctl/client.go

```diff
@@ -131,15 +131,13 @@ type UpgradeInput struct {
 
 // Upgrade calls clusterctl upgrade apply with the list of providers defined in the local repository.
 func Upgrade(ctx context.Context, input UpgradeInput) {
-	if len(input.ClusterctlVariables) > 0 {
```

Comment:
same below

Reply:
I suggested to drop it so that we always have a copy. I think this just makes it a bit easier to see in everyone of those tests which clusterctl-config file was used for the cluster.

But no objections if we want to add the if in both cases

Reply:
Fine for me either way

Reply:
Adding the if to both :-)